trollsched/tests/test_pass_outlines.py

```diff
@@ -0,0 +1,113 @@
+                          0.67750383, 0.68149934, 0.68493465, 0.68794744, 0.69063434,
+                          0.69306561, 0.6952939 , 0.6973597 , 0.69929488, 0.7011251 ,
+                          0.7028715 , 0.7045518 , 0.70618126, 0.70777322, 0.70933965,
+                          0.7108915 , 0.71243895, 0.71399159, 0.71555848, 0.71714804,
+                          0.71876762, 0.72042255, 0.72211402, 0.72383449, 0.72555719,
+                          0.72721033, 0.72860415, 0.72917251, 0.95338892, 1.04189611,
+                          1.12986537, 1.21694627, 1.30234019, 1.38380771, 1.45301036],
+                         "float64")
+
+
+def test_get_shapely_polygon_from_lonlat():
+    """Test getting a polygon from a swath contour polygon."""
+    sat_poly = get_shapely_polygon_from_lonlat(CONT_POLY_LON, CONT_POLY_LAT)
+
+    assert isinstance(sat_poly, shapely.geometry.polygon.Polygon)
+    assert sat_poly.is_valid
+    assert sat_poly.is_simple
+    assert sat_poly.is_closed is False
+    np.testing.assert_almost_equal(sat_poly.length, 649.84664, decimal=5)
+
+
+@pytest.mark.usefixtures("fake_noaa20_viirs_pass_instance")
+def test_create_shapefile_filename(fake_noaa20_viirs_pass_instance):
+    """Test creating the shapefile filename from a NOAA-20/VIIRS trollsched.satpass instance."""
+    result = create_shapefile_filename(fake_noaa20_viirs_pass_instance)
+    assert result == "viirs_NOAA-20_202409170125_202409170140_outline.shp"
+
+
+@pytest.mark.usefixtures("fake_noaa20_viirs_pass_instance")
+def test_create_shapefile_from_pass(fake_noaa20_viirs_pass_instance, tmp_path):
+    """Test create a shapefile from a fake NOAA-20/VIIRS pass."""
+    tmp_filename = tmp_path / "fake_noaa20_viirs_shapefile.shp"
+    create_shapefile_from_pass(fake_noaa20_viirs_pass_instance, tmp_filename)
+    assert tmp_filename.exists()
+    nfiles = len([f for f in tmp_filename.parent.glob("fake_noaa20_viirs_shapefile.*")])
+    assert nfiles == 5
+
+
+@pytest.mark.usefixtures("fake_long_tle_file")
+@pytest.mark.usefixtures("fake_xml_schedule_file")
+def test_shapefiles_from_schedule_xml_requests(fake_xml_schedule_file, fake_long_tle_file, tmp_path):
+    """Test create shapefiles from an xml schedule request file."""
+    output_dir = tmp_path / "results"
+    output_dir.mkdir()
+    satellites = ["Suomi NPP", "NOAA 18", "NOAA-20"]
+    shapefiles_from_schedule_xml_requests(str(fake_xml_schedule_file), satellites,
+                                          str(fake_long_tle_file), str(output_dir))
+
+    n18files = [str(name) for name in output_dir.glob("avhrr_NOAA-18_202412022201_202412022216_outline.*")]
+    assert len(n18files) == 5
+    nppfiles = [str(name) for name in output_dir.glob("viirs_Suomi-NPP_202412022238_202412022249_outline.*")]
+    assert len(nppfiles) == 5
+    n20files = [str(name) for name in output_dir.glob("viirs_NOAA-20_202412022301_202412022314_outline.*")]
+    assert len(n20files) == 5
```

Comment:
Can you make a copy of this test that calls `run` with args instead? that way, `run` and `parse_args` are tested too…

Reply:
Ok, good point, fixed!